compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -210,21 +212,21 @@ impl CodegenCx<'ll, 'tcx> {
 
         debug!("get_static: sym={} instance={:?}", sym, instance);
 
-        let g = if let Some(def_id) = def_id.as_local() {
-            let id = self.tcx.hir().local_def_id_to_hir_id(def_id);
+        let g = if let Some(local_def_id) = def_id.as_local() {
+            let id = self.tcx.hir().local_def_id_to_hir_id(local_def_id);
             let llty = self.layout_of(ty).llvm_type(self);
             // FIXME: refactor this to work without accessing the HIR
             let (g, attrs) = match self.tcx.hir().get(id) {
-                Node::Item(&hir::Item { attrs, span, kind: hir::ItemKind::Static(..), .. }) => {
+                Node::Item(&hir::Item { attrs, kind: hir::ItemKind::Static(..), .. }) => {
                     if let Some(g) = self.get_declared_value(sym) {
                         if self.val_ty(g) != self.type_ptr_to(llty) {
-                            span_bug!(span, "Conflicting types for static");
+                            span_bug!(self.tcx.def_span(def_id), "Conflicting types for static");
```

Comment:
is there a reason you aren't using the `Item` span here? That seems easier to me

Reply:
At the moment, it is, yes. Consider this change a step towards #43088 :)

Reply:
Also I actually dont know, by the way, if these are identical, so it's kind of a "let's see what the tests think" kind of change.

Reply:
the tests won't think anything here will they? This is a `span_bug` so we should new reach this in tests

Reply:
GitHub needs a reaction that indicates how blind I can be :)